presto-accumulo/src/main/java/io/prestosql/plugin/accumulo/model/Field.java

```diff
@@ -266,279 +266,92 @@ private static boolean equals(Block block1, Block block2)
-                else if (Types.isMapType(valueType)) {
-                    values.append(
-                            new Field(
-                                    AccumuloRowSerializer.getBlockFromMap(valueType, (Map<?, ?>) entry.getValue()),
-                                    valueType))
-                            .append(',');
-                }
-                else {
-                    values.append(new Field(entry.getValue(), valueType)).append(',');
-                }
-            }
-
-            keys.deleteCharAt(keys.length() - 1).append(']');
-            values.deleteCharAt(values.length() - 1).append(']');
-            return builder.append(keys).append(", ").append(values).append(")").toString();
-        }
-
-        // Validate the object is the given type
-        if (type.equals(BIGINT) || type.equals(BOOLEAN) || type.equals(DOUBLE) || type.equals(INTEGER) || type.equals(REAL) || type.equals(TINYINT) || type.equals(SMALLINT)) {
-            return value.toString();
-        }
-        if (type.equals(DATE)) {
-            return "DATE '" + value.toString() + "'";
-        }
-        if (type.equals(TIME)) {
-            return "TIME '" + value.toString() + "'";
-        }
-        if (type.equals(TIMESTAMP)) {
-            return "TIMESTAMP '" + value.toString() + "'";
-        }
-        if (type.equals(VARBINARY)) {
-            return "CAST('" + new String((byte[]) value, UTF_8).replaceAll("'", "''") + "' AS VARBINARY)";
-        }
-        if (type instanceof VarcharType) {
-            return "'" + value.toString().replaceAll("'", "''") + "'";
-        }
-        throw new PrestoException(NOT_SUPPORTED, "Unsupported PrestoType " + type);
+        return toStringHelper(this)
+                .add("value", value)
+                .add("type", type)
+                .toString();
     }
 
     /**
-     * Does it's damnedest job to convert the given object to the given type.
+     * Convert Presto native value (stack representation) of given type to Accumulo equivalent.
      *
      * @param value Object to convert
      * @param type Destination Presto type
-     * @return Null if null, the converted type of it could convert it, or the same value if it is fine just the way it is :D
-     * @throws PrestoException If the given object is not any flavor of the given type
      */
-    private static Object cleanObject(Object value, Type type)
+    private static Object convert(Object value, Type type)
     {
         if (value == null) {
             return null;
         }
 
+        checkArgument(Primitives.wrap(type.getJavaType()).isInstance(value), "Invalid representation for %s: %s [%s]", type, value, value.getClass().getName());
+
         // Array? Better be a block!
         if (Types.isArrayType(type)) {
-            if (!(value instanceof Block)) {
-                throw new PrestoException(FUNCTION_IMPLEMENTATION_ERROR, "Object is not a Block, but " + value.getClass());
-            }
+            // Block
```

Comment:
Should we enforce the types here?
```java
return (Block) value;

Reply:
I'd like to do this, but intellij frowns on "unnecessary cast". (which i dont want to ignore in general, although here it's incorrect).
This is, however, enforced by the `checkArgument(Primitives.wrap(type.getJavaType()).isInstance(value),` check above.